packages/core/src/odata-common/entity-builder.spec.ts

```diff
@@ -206,9 +206,19 @@ describe('EntityBuilder', () => {
       const entity = TestEntity.builder().fromJson(entityJson);
 
       expect(entity).toStrictEqual(expectedEntity);
-      expect(entity.toSingleLink.getCustomFields()).toEqual(
-        expectedEntity.toSingleLink.getCustomFields()
+      expect(entity.toSingleLink!.getCustomFields()).toEqual(
+        expectedEntity.toSingleLink!.getCustomFields()
       );
     });
+
+    it('should build an entity from json with one-to-one navigation properties being null', () => {
+      const entity = TestEntity.builder().fromJson({
+        toSingleLink: null
+      });
+      const expectedEntity = TestEntity.builder()
+        .toSingleLink(null)
```

Comment:
undefined should also work right? Add a similar test for undefined if this is the case.